File: courses-back/authentication/src/main/java/com/company/courses/authentication/model/Account.java
package com.company.courses.authentication.model;

import com.company.courses.authentication.model.enums.AccountStatusEnum;
import com.company.courses.authentication.model.enums.AccountTypeEnum;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Document("accounts")
public class Account {

    private String id;
    private String authId;
    private String userId;
    private LocalDateTime createdDate;
    private LocalDateTime updatedDate;
    private AccountStatusEnum status;
    private AccountTypeEnum type;

    public Account(String authId, String userId, AccountTypeEnum type) {
        this.authId = authId;
        this.userId = userId;
        this.createdDate = LocalDateTime.now();
        this.status = AccountStatusEnum.ACTIVE;
        this.type = type;
        this.createdDate = LocalDateTime.now();
    }

    public String getId() {
        return this.id;
    }

    public String getUserId() {
        return this.userId;
    }

    public LocalDateTime getCreatedDate() {
        return this.createdDate;
    }

    public LocalDateTime getUpdatedDate() {
        return this.updatedDate;
    }

    public AccountStatusEnum getStatus() {
        return this.status;
    }

    public void setStatus(AccountStatusEnum status){
        this.status = status;
    }

    public void setType(AccountTypeEnum type) {
        this.type = type;
    }
}

File: courses-back/accounts/Models/Account.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace accounts.Models
{
    public class Account
    {

        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? Id {  get; set; }
        public string UserId { get; set; } = null!;
        public string AuthId { get; set; } = null!;
        public DateTime CreatedDate { get; set; }
        public DateTime UpdateDate { get; set; }
        public string Status { get; set; } = null!;
        public string Type { get; set; } = null!;
    }
}
